Source/WebCore/page/csp/ContentSecurityPolicy.cpp

```diff
@@ -776,6 +776,42 @@ AllowTrustedTypePolicy ContentSecurityPolicy::allowTrustedTypesPolicy(const Stri
         : details;
 }
 
+// https://w3c.github.io/trusted-types/dist/spec/#does-sink-require-trusted-types
+bool ContentSecurityPolicy::requireTrustedTypesForSinkGroup(const String& sinkGroup) const
+{
+    bool required = false;
+    for (auto& policy : m_policies) {
+        if (policy->requiresTrustedTypesForScript() && sinkGroup == "script"_s) {
+            required = true;
+            break;
+        }
+    }
+    return required;
+}
+
+// https://w3c.github.io/trusted-types/dist/spec/#should-block-sink-type-mismatch
+bool ContentSecurityPolicy::allowMissingTrustedTypesForSinkGroup(const String& stringContext, const String& sink, const String& sinkGroup, const String& source) const
+{
+    return allOf(m_policies, [&](auto& policy) {
+        bool isAllowed = true;
+        if (policy->requiresTrustedTypesForScript() && sinkGroup == "script"_s) {
+            if (!policy->isReportOnly())
+                isAllowed = false;
+
+            String consoleMessage = makeString(policy->isReportOnly() ? "[Report Only] " : "",
+                "This requires a "_s, stringContext, " value else it violates the following Content Security Policy directive: \"require-trusted-types-for 'script'\""_s);
+
+            const ContentSecurityPolicyDirectiveList& list = *policy.get();
+
+            String sourceURL;
```

Comment:
Seems a little strange to use a local variable for this. Just pass `nullString()` instead?